src/assets/TokenRatesController.ts

```diff
@@ -269,7 +274,32 @@ export class TokenRatesController extends BaseController<
     chainSlug: string,
     query: string,
   ): Promise<CoinGeckoResponse> {
-    return handleFetch(CoinGeckoApi.getTokenPriceURL(chainSlug, query));
+    try {
+      const coinGeckoResponse: CoinGeckoResponse = await handleFetch(
+        CoinGeckoApi.getTokenPriceURL(chainSlug, query),
+      );
+      return coinGeckoResponse;
+    } catch {
+      return {};
```

Comment:
It looks like any errors in the `try` block are getting swallowed here, and in `checkIsSupportedVsCurrency`. This could hide bugs. Is there a specific error you wanted to suppress here?

Reply:
No this was a lingering remnant of some debugging I was doing while implementing. But I guess since we're here, do we want to surface errors from this API call? We suppress them for the call in `fetchSupportedChains` (just above here). I suppose we could log it somewhere?  Is that what you are recommending?

Reply:
We should always propagate all errors that we don't know how to handle (unless we have a very good reason not to at least). In this case I don't think there are any errors we know how to handle in this function.

In this case we know and expect this to throw an error sometimes, like if the server is down or the user isn't connected to the internet. So bubbling those cases up to be reported to Sentry isn't useful - that would just flood our Sentry queue with non-issues. But this is still probably the wrong layer to catch it, because this function could be called from the UI where we'd want the UI to know the operation failed.

What I would do to improve this error handling is:

* Document the anticipated errors (like network errors and 5xx server responses) with a `@throws` tag.
* Catch them in the polling function and log them to the console, stopping them from propagating further.
* Audit the other places where this function is called, and evaluate how best to deal with it in each place.

That's well outside the scope of this PR though. Right now we catch and ignore all polling errors, which is not great (I would like to abolish `safelyExecute` completely to be honest), but we don't have to worry about polling network errors being sent to Sentry at least.